scraper.rb

```diff
@@ -47,11 +47,41 @@ class MemberPage < Scraped::HTML
     url.to_s
   end
 
+  field :start_date do
+    date_from(
+      recent_electoral_history
+                  .scan(/(?:Elected on|Oath of Allegiance|casual elections on)[^:]*\:\s*(\d{1,2}\.\d{1,2}\.\d{2,4})/)
+                  .flatten
+                  .last
+    )
+  end
+
+  field :end_date do
```

Comment:
There's a little too much duplication here with `start_date`. I think, more generally, that doing this with a regex is probably the wrong approach. Rather than throwing away all the underlying structure by collapsing the whole section to `.text`, I think it might be simpler, more robust, **and** more in keeping with the principles of providing as much data as possible from the page, to turn this entire History section into a data structure, perhaps along the lines of:

```
[
  { 
    "Eleventh Parliament": {
      "Elected on": "2008-03-12",
      "Re-Appointed Deputy Speaker": "2008-05-10",
      "Resigned from Deputy Speaker": "2010-05-07",
      "Dissolution of Parliament": "2013-01-07",
    }
  },
  {
    "Twelfth Parliament": {
      "Elected on": "2013-03-11",
      "Oath of Allegiance as MP": "2013-04-06",
      "Sworn in as Minister for Home Affairs and National Security on": "2014-12-09",
    }
  }
]
```